src/main/java/com/limechain/transaction/TransactionPool.java

```diff
@@ -0,0 +1,46 @@
+package com.limechain.transaction;
+
+import com.limechain.transaction.dto.Extrinsic;
+import com.limechain.transaction.dto.ValidTransaction;
+import com.limechain.utils.HashUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class TransactionPool {
+
+    private final Map<byte[], ValidTransaction> transactions;
+
+    public TransactionPool() {
+        transactions = new HashMap<>();
+    }
```

Comment:
You can move the initialization of the map to the field declaration and delete the constructor entirely